src/main/java/frc/robot/RobotContainer.java

```diff
@@ -31,9 +43,16 @@ public RobotContainer() {
         inputChooser.setDefaultOption("Single F310", new ControllerInput());
 
         drive.setDefaultCommand(driveCommand);
+        elevator.setDefaultCommand(elevatorCommand);
     }
 
-    private void configureBindings() {}
+    private void configureBindings() {
+        endEffectorExAlgaeInCoralTrigger = new Trigger(() -> getInput().getEndEffectorIn());
+        endEffectorInAlgaeExCoralTrigger = new Trigger(() -> getInput().getEndEffectorOut());
+
+        endEffectorExAlgaeInCoralTrigger.whileTrue(algaeOut);
+        endEffectorInAlgaeExCoralTrigger.and(endEffectorExAlgaeInCoralTrigger.whileFalse(algaeIn));
```

Comment:
Not sure we need to check the `endEffectorExAlgaeInCoralTrigger` here.
This is handled by the command framework, given the following:
- Both the `algaeOut` and `algaeIn` commands require the end effector
- [Trigger#whileTrue](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/button/Trigger.html#whileTrue(edu.wpi.first.wpilibj2.command.Command)) will schedule the command only once, when it first becomes true. It will not reschedule the command.

This means that whichever trigger is triggered last will win.
For example,
1. The driver commands algae out.
2. The `endEffectorExAlgaeInCoralTrigger` trigger schedules the `algaeOut` command.
3. The driver commands algae in.
4. The `endEffectorInAlgaeExCoralTrigger` schedules the `algaeIn` command.
5. The scheduler notices that there is already a command running on the end effector. It interrupts the `algaeOut` command and starts running the `algaeIn` command instead.

```suggestion
        endEffectorInAlgaeExCoralTrigger.whileTrue(algaeIn);
```